test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -748,25 +748,38 @@ private static void deleteAllNodeShutdownMetadata() throws IOException {
     }
 
     protected static void wipeAllIndices() throws IOException {
+        final String[] indicesToIgnore = new String[] {
+            // ignore ilm history which can pop up after deleting all data streams but shouldn't interfere
+            ".ds-ilm-history-*"
+        };
+
         boolean includeHidden = minimumNodeVersion().onOrAfter(Version.V_7_7_0);
         try {
-            //remove all indices except ilm history which can pop up after deleting all data streams but shouldn't interfere
-            final Request deleteRequest = new Request("DELETE", "*,-.ds-ilm-history-*");
+            final Request deleteRequest = new Request("DELETE", "*,-" + String.join(",-", indicesToIgnore));
             deleteRequest.addParameter("expand_wildcards", "open,closed" + (includeHidden ? ",hidden" : ""));
-            RequestOptions allowSystemIndexAccessWarningOptions = RequestOptions.DEFAULT.toBuilder()
-                .setWarningsHandler(warnings -> {
-                    if (warnings.size() == 0) {
-                        return false;
-                    } else if (warnings.size() > 1) {
-                        return true;
-                    }
-                    // We don't know exactly which indices we're cleaning up in advance, so just accept all system index access warnings.
-                    final String warning = warnings.get(0);
-                    final boolean isSystemIndexWarning = warning.contains("this request accesses system indices")
-                        && warning.contains("but in a future major version, direct access to system indices will be prevented by default");
-                    return isSystemIndexWarning == false;
-                }).build();
-            deleteRequest.setOptions(allowSystemIndexAccessWarningOptions);
+
+            if (minimumNodeVersion().onOrAfter(Version.CURRENT)) {
+                // feature reset deletes system indices
+                final Request postRequest = new Request("POST", "/_features/_reset");
+                adminClient().performRequest(postRequest);
```

Comment:
We might want to check the response, since it's possible for this request to only partially succeed. That shouldn't happen in a test, but if it does, I'd rather get a stack trace pointing here than have another test fail with a really unclear error.

Reply:
+100 on logging both exceptions and partial failures here

> That shouldn't happen in a test

But it will, especially in PR builds, because somebody will make a false assumption about how their code changes interact with feature reset.  The feature reset API will become one of the most heavily tested APIs in the product when this PR is merged.  All the other REST tests that exist will exercise an enormous number of edge cases in feature reset (which is good, but we should plan for failures).

Reply:
I've updated this so that we throw an error when feature reset fails. I expected that the HTTP Entity would be the full response message we return on an error, but it's just a root cause message. But at least now we have a message that indicates a problem with resetting features.